src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
1. It's fine to block out file paths. Assert that the path starts with http.
2. All three are cool. There should not be any repetition if the class stores base and relative paths instead of base and full URL as it does now. The class should not be tied to Config.APP_URL in anyway. The only way it can store Config.APP_URL is if it is fed via a constructor i.e. 
```java
new Url("http://www.google.com/page").getBaseUrl() = "http://www.google.com";
```

| `:Url` |
--------------------
|`base: http://www.google.com` <br> `relativePath: page` |
-------------------
```java
new Url(Config.APP_URL + "/page").getBaseUrl() = Config.APP_URL;
```

| `:Url` |
--------------------
|`base: Config.APP_URL` <br> `relativePath: page` |
-------------------